From d5a2b9b917312c64e145c7427713c9b1865446be Mon Sep 17 00:00:00 2001
From: Sergio Pena <sergio.pena@cloudera.com>
Date: Mon, 22 Feb 2016 11:15:20 -0600
Subject: [PATCH 0509/1164] CLOUDERA-BUILD: CDH-37380: Hive itests/ project
 build fails when using an incorrect
 spark-latest.tar.gz file

Change-Id: I2f7a5fb71027b6c38b74e30b0807e8d3de265031
---
 itests/pom.xml |   16 ++++++++++++----
 1 file changed, 12 insertions(+), 4 deletions(-)

diff --git a/itests/pom.xml b/itests/pom.xml
index e5c68a0..d9a36fd 100644
--- a/itests/pom.xml
+++ b/itests/pom.xml
@@ -83,16 +83,24 @@
                     finalName=$2
                     tarName=$(basename $url)
                     rm -rf $BASE_DIR/$finalName
-                    if [[ ! -f $DOWNLOAD_DIR/$tarName ]]
-                    then
-                     curl -Sso $DOWNLOAD_DIR/$tarName $url
+                    if [[ ! -f $DOWNLOAD_DIR/$tarName ]]; then
+                     curl -Sso $DOWNLOAD_DIR/$tarName $url || return 1
+                    else
+                      content=$(tar --exclude='*/*' -tf "$DOWNLOAD_DIR/$tarName")
+                      if ! echo "$content" | grep "spark-${spark.version}" >/dev/null 2>/dev/null; then
+                        curl -Sso $DOWNLOAD_DIR/$tarName $url || return 1
+                      fi
                     fi
                     tar -zxf $DOWNLOAD_DIR/$tarName -C $BASE_DIR
                     mv $BASE_DIR/spark-${spark.version} $BASE_DIR/$finalName
                   }
                   mkdir -p $DOWNLOAD_DIR
                   CDH_VERSION=$(echo ${spark.version}|cut -d- -f2)
-                  download "http://repos.jenkins.cloudera.com/${CDH_VERSION}-static/cdh/5/spark-latest.tar.gz" "spark"
+                  REPO_URL="http://repos.jenkins.cloudera.com/${CDH_VERSION}-static/cdh/5/spark-latest.tar.gz"
+                  if ! download "$REPO_URL" "spark"; then
+                    echo "ERROR: Cannot download spark-latest.tar.gz from $REPO_URL"
+                    exit 1
+                  fi
                   cp -f $HIVE_ROOT/data/conf/spark/log4j.properties $BASE_DIR/spark/conf/
                 </echo>
               </target>
-- 
1.7.9.5

